sources/ClangSharpPInvokeGenerator/Program.cs

```diff
@@ -793,435 +872,292 @@ private static void ParseKeyValuePairs(IEnumerable<string> keyValuePairs, List<s
-                    arity: ArgumentArity.OneOrMore
-                );
-            }
-            rootCommand.AddOption(s_configOption);
+            return s_configOption = new Option<string[]>(
+                aliases: new string[] { "--config", "-c" },
+                description: "A configuration option that controls how the bindings are generated. Specify 'help' to see the available options.",
+                getDefaultValue: Array.Empty<string>
+            );
         }
 
-        private static void AddDefineMacroOption(RootCommand rootCommand)
+        private static Option<string[]> GetDefineMacroOption()
         {
-            var option = new Option(
+            return new Option<string[]>(
                 aliases: new string[] { "--define-macro", "-D" },
                 description: "Define <macro> to <value> (or 1 if <value> omitted).",
-                argumentType: typeof(string),
-                getDefaultValue: Array.Empty<string>,
-                arity: ArgumentArity.OneOrMore
+                getDefaultValue: Array.Empty<string>
             );
-
-            rootCommand.AddOption(option);
         }
 
-        private static void AddExcludeOption(RootCommand rootCommand)
+        private static Option<string[]> GetExcludeOption()
         {
-            var option = new Option(
+            return new Option<string[]>(
                 aliases: new string[] { "--exclude", "-e" },
                 description: "A declaration name to exclude from binding generation.",
-                argumentType: typeof(string),
-                getDefaultValue: Array.Empty<string>,
-                arity: ArgumentArity.OneOrMore
+                getDefaultValue: Array.Empty<string>
             );
-
-            rootCommand.AddOption(option);
         }
 
-        private static void AddFileOption(RootCommand rootCommand)
+        private static Option<string[]> GetFileOption()
         {
-            var option = new Option(
+            return new Option<string[]>(
                 aliases: new string[] { "--file", "-f" },
                 description: "A file to parse and generate bindings for.",
-                argumentType: typeof(string),
-                getDefaultValue: Array.Empty<string>,
-                arity: ArgumentArity.OneOrMore
+                getDefaultValue: Array.Empty<string>
             );
-
-            rootCommand.AddOption(option);
         }
 
-        private static void AddFileDirectoryOption(RootCommand rootCommand)
+        private static Option<string> GetFileDirectoryOption()
         {
-            var option = new Option(
+            return new Option<string>(
                 aliases: new string[] { "--file-directory", "-F" },
                 description: "The base path for files to parse.",
-                argumentType: typeof(string),
-                getDefaultValue: () => string.Empty,
-                arity: ArgumentArity.ExactlyOne
```

Comment:
This is now defaulted based on the generic type of the option (which is forwarded to the option's argument). 
It can be overridden by specifying a specific value for the `Arity` property, however it appeared that the options here were all following the defaults.

You can see the default Arity for different types here: https://github.com/dotnet/command-line-api/blob/main/src/System.CommandLine/ArgumentArity.cs#L144-L168